### ISSUE TITLE
IOS native can't handle parallel connection from node

### ISSUE THREAD

**Issue by chapko** [thaliproject/Thali_CordovaPlugin#1845]
I added more tests to the `master_chapko_minidatashifttest` branch.

The simplest one ('Can shift data via parallel connections') does the same as 'Can shift data' but using 3 parallel non-tls connections. Server receives data from 3 client sockets, sends data back, but only 1 of the 3 connections receives it back from server.

Logs:
- [device1.log](https://gist.github.com/chapko/a33de09d371796693b0bec462cafae15/raw/fe492c28c46c3201369b68026964880d60b2a5c9/device1.log)
- [device2.log](https://gist.github.com/chapko/a33de09d371796693b0bec462cafae15/raw/fe492c28c46c3201369b68026964880d60b2a5c9/device2.log)

**Comment by dersim-davaod** [comment]
Updated logs:
[Device 1](https://gist.github.com/dersim-davaod/090560941fe77895aee77e7190ab2d67)
[Device 2](https://gist.github.com/dersim-davaod/8859c2b6ef73873db1ef5b43591b2e1b)

**Comment by enricogior** [comment]
The test may be failing for a problem with the test itself.
I've sent the detailed information to Dersim.
In a nutshell: the server is calling socket.end() after a socket.write(), the problem is that libuv closes the socket before the native layer has sent the msg. That may not happen on the desktop or Android because of the thinner native layer that manage to send the msg before libuv closes the socket.
The proposed fix is to avoid closing the socket on the server and let the client decide when it's time to close it.

**Comment by enricogior** [comment]
I did more investigation on the issue, it turned out that no matter when the server issues a `socket.end`, things will break.
I modified the test so that the server sends back the received message and closes the socket 250ms later.
I see all the three massages been received back by the clients, then the first `socket.end` is executed on the server side and the first client receives the socket 'on end' event correctly. After that the connection between the server and the other two clients is broken and no other socket 'end' events are received by the clients.
It looks like the native layer/MPCF may have an incorrect behavior after the server closes the first socket, treating all following socket operations, coming from the server, to not been forwarded to the other peer.

I also tried a different test: instead of closing the sockets on the server, the sockets are closed on the client in the "on data" event after the received data is compared to the expected data. In that case everything works as expected. Closing the socket on the client doesn't cause problems to the other sockets that are still processing data between the two peers.
This second test was done to make sure the problem is not occurring on both ends but only when is the server to close the socket.
